riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/DefaultRiptideRegistrar.java

```diff
@@ -168,10 +170,11 @@ private String registerExecutor(final String id, final Client client) {
         }
 
         if (client.getTracing().getEnabled()) {
+            BeanReference tracerRef = registry.findRef(null, Tracer.class).orElseThrow(() -> new NoSuchBeanDefinitionException(Tracer.class));
```

Comment:
`findRef` still searches by name. It's just using the given class to construct a name. There is no easy way to autowire by type when dealing with a bean definition registry. It's one of the downsides of how the auto configuration is implemented right now.

Reply:
thanks for pointing out :/